modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/ObjectDefinitionExportImportTest.java

```diff
@@ -65,6 +81,128 @@ public void testExportImportObjectDefinition() throws Exception {
+			).put(
+				"externalReferenceCode", "TESTOBJECTDEFINITION1"
+			).put(
+				"name", "TestObjectDefinition1"
+			).put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 0
+				).put(
+					"label", "approved"
+				).put(
+					"label_i18n", "Approved"
+				)
+			).toString();
+
+		testExportImportJSONString(
+			publishedObjectDefinitionJSONString,
+			publishedObjectDefinitionJSONString, "TESTOBJECTDEFINITION1",
+			"TestObjectDefinition1");
+
+		// Cannot modify the status of published object definition
+
+		Class<?> clazz = getClass();
+
+		testFailedImportJSONString(
+			_jsonFactory.createJSONObject(
+				publishedObjectDefinitionJSONString
+			).put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 2
+				).put(
+					"label", "draft"
+				).put(
+					"label_i18n", "Draft"
+				)
+			).toString(),
+			StringUtil.read(
+				clazz.getResourceAsStream(
+					"dependencies/test-object-definition.draft.error-message." +
+						"json")),
+			"TESTOBJECTDEFINITION1", "TestObjectDefinition1");
+
+		// Import bound object definitions one by one
+
+		String objectDefinitionRootJSONString = _jsonFactory.createJSONObject(
+			_defaultObjectDefinitionJSONString
+		).put(
+			"externalReferenceCode", "TESTOBJECTDEFINITION2"
+		).put(
+			"name", "TestObjectDefinition2"
+		).put(
+			"objectRelationships",
+			JSONUtil.put(
+				_jsonFactory.createJSONObject(
+					_defaultObjectRelationshipJSONString
+				).put(
+					"edge", true
+				).put(
+					"name", "objectRelationship1"
+				).put(
+					"objectDefinitionExternalReferenceCode1",
+					"TESTOBJECTDEFINITION2"
+				).put(
+					"objectDefinitionExternalReferenceCode2",
+					"TESTOBJECTDEFINITION3"
+				).put(
+					"objectDefinitionName2", "TESTOBJECTDEFINITION3"
+				).put(
+					"type", "oneToMany"
+				))
+		).put(
+			"rootObjectDefinitionExternalReferenceCode", "TESTOBJECTDEFINITION2"
+		).toString();
+
+		testExportImportJSONString(
+			objectDefinitionRootJSONString, objectDefinitionRootJSONString,
+			"TESTOBJECTDEFINITION2", "TestObjectDefinition2");
+
+		Assert.assertNotNull(
+			_objectDefinitionResource.
+				getObjectDefinitionByExternalReferenceCode(
+					"TESTOBJECTDEFINITION3"));
+
+		String objectDefinitionBoundJSONString = _jsonFactory.createJSONObject(
+			_defaultObjectDefinitionJSONString
+		).put(
+			"externalReferenceCode", "TESTOBJECTDEFINITION3"
+		).put(
+			"name", "TestObjectDefinition3"
+		).put(
+			"rootObjectDefinitionExternalReferenceCode", "TESTOBJECTDEFINITION2"
+		).toString();
+
+		JSONObject expectedObjectDefinitionBoundJSONObject =
+			_jsonFactory.createJSONObject(objectDefinitionBoundJSONString);
+
+		expectedObjectDefinitionBoundJSONObject.put(
+			"objectFields",
+			expectedObjectDefinitionBoundJSONObject.getJSONArray(
+				"objectFields"
+			).put(
+				_jsonFactory.createJSONObject(
+				).put(
+					"name", "r_objectRelationship1_c_testObjectDefinition2Id"
+				)
+			));
+
```

Comment:
This scenario is easier to understand, so I think we can leave it this way, but feel free if you want to try to fill in the information in the import data.